hyperspy/misc/machine_learning/import_sklearn.py

```diff
@@ -27,3 +27,5 @@
         sklearn_installed = True
 except ImportError:
     sklearn_installed = False
+finally:
+    warnings.simplefilter("default")
```

Comment:
That's what I thought, but adding it fixed the issue. However this may be because, as you pointed out somewhere else, setting it to "default" has the counterintuitive consequence of printing the DeprecationWarnings which are not printed by default...